src/coffea/nanoevents/methods/fcc.py

```diff
@@ -0,0 +1,452 @@
+        - The Particleidx0.index contains info about the parents
+        """
+        ranges = index_range(self.parents.begin, self.parents.end)
+        return awkward.values_astype(
+            map_index_to_array(self._events().Particleidx0.index, ranges, axis=2),
+            "int64",
+        )
+
+    @get_parents_index.dask
+    def get_parents_index(self, dask_array):
+        """
+        Obtain the indexes of the parents of each and every MCParticle
+        - The output is a doubly nested awkward array
+        - Needs the presence of Particleidx0 collection
+        - The Particleidx0.index contains info about the parents
+
+        Note: Seems like all the functions need to mapped manually
+        """
+        ranges = dask_awkward.map_partitions(
+            index_range, dask_array.parents.begin, dask_array.parents.end
+        )
+        daughters = dask_awkward.map_partitions(
+            map_index_to_array, dask_array._events().Particleidx0.index, ranges, axis=2
+        )
+        return awkward.values_astype(daughters, "int32")
+
+    @dask_property
+    def get_parents(self):
+        """
+        Obtain the actual MCParticle parents to each and every MCParticle
+        - The output is a doubly nested awkward array
+        - Needs the presence of Particleidx0 collection
+        - The Particleidx0.index contains info about the parents
+        """
+        return map_index_to_array(self, self.get_parents_index, axis=2)
+
+    @get_parents.dask
+    def get_parents(self, dask_array):
+        """
+        Obtain the actual MCParticle parents to each and every MCParticle
+        - The output is a doubly nested awkward array
+        - Needs the presence of Particleidx0 collection
+        - The Particleidx0.index contains info about the parents
+        """
+        return map_index_to_array(dask_array, dask_array.get_parents_index, axis=2)
+
+
+_set_repr_name("MCParticle")
+behavior.update(awkward._util.copy_behaviors(MomentumCandidate, MCParticle, behavior))
+
+MCParticleArray.ProjectionClass2D = vector.TwoVectorArray  # noqa: F821
+MCParticleArray.ProjectionClass3D = vector.ThreeVectorArray  # noqa: F821
+MCParticleArray.ProjectionClass4D = MCParticleArray  # noqa: F821
+MCParticleArray.MomentumClass = vector.LorentzVectorArray  # noqa: F821
+
+
+@awkward.mixin_class(behavior)
+class ReconstructedParticle(MomentumCandidate, base.NanoCollection):
+    """Reconstructed particle"""
+
+    def match_collection(self, idx):
+        """Returns matched particles"""
+        return self[idx.index]
+
+    @dask_property
+    def matched_gen(self):
+        sel = awkward.broadcast_arrays(True, self)[0]
+        index = self._events().MCRecoAssociations.reco_mc_index[:, :, 1]
+        return self._events().Particle[index[sel]]
+
+    @matched_gen.dask
+    def matched_gen(self, dask_array):
+        sel = awkward.broadcast_arrays(True, dask_array)[0]
+        index = dask_array._events().MCRecoAssociations.reco_mc_index[:, :, 1]
```

Comment:
Same issue with the indexing through cuts. I had most of an example together in #822.